hashicorp-vault/main.tf

```diff
@@ -0,0 +1,224 @@
+data "aws_eks_cluster" "cluster" {
+  name = var.cluster_name
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = var.cluster_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.cluster.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+    token                  = data.aws_eks_cluster_auth.cluster.token
+  }
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster.token
+}
+
+resource "kubernetes_namespace" "this" {
+  count = var.chart_create_namespace ? 1 : 0
+  metadata {
+    annotations = {
+      name = var.chart_namespace
+    }
+
+    name = var.chart_namespace
+  }
+}
+
+resource "aws_kms_key" "this" {
+  count                   = var.s3_create_bucket ? 1 : 0
+  description             = "Key for vault s3 bucket"
+  deletion_window_in_days = 10
+}
+
+
+resource "aws_s3_bucket" "this" {
+  count  = var.s3_create_bucket ? 1 : 0
+  bucket = var.s3_bucket_name
+  acl    = "private"
+
+  tags = var.tags
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        kms_master_key_id = aws_kms_key.this.0.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+}
+
+resource "aws_iam_user_policy" "this" {
+  count = var.s3_storage ? 1 : 0
+  name  = "VaultUserPermissions"
+  user  = aws_iam_user.this.0.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "s3:*"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
```

Comment:
why do I need access to all s3 buckets? Too wide a set of rights, you need to restrict access only to your s3 bucket